packages/pluginsdk-peerdeps/sync-stage.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+# This script creates a temporary directory and scaffolds a new plugin.
+# It updates that scaffolded plugin's regular dependencies using the sync-deck.js script.
+
+set -e
+
+SCRIPTDIR=$(cd "$(dirname "$0")"; pwd -P)
+
+cd $(mktemp -d)
+SCAFFOLDDIR=${PWD}
+
+echo "${SCAFFOLDDIR}" > "${SCRIPTDIR}/.scaffolddir"
+
+echo "pluginsdk-peerdeps: Scaffolding a new plugin into $SCAFFOLDDIR"
+"$SCRIPTDIR"/../pluginsdk/scripts/scaffold.js --directory . --name scaffold
+echo "[pluginsdk-peerdeps] New plugin scaffolded into $SCAFFOLDDIR"
+echo "node_modules" > .gitignore
```

Comment:
We need to have the newly created plugin directory appended to `SCAFFOLDDIR` before we run this one?
On a related note, I feel that it is better to use the defaults to scaffold the plugin without any user prompts and would also help here in hardcoding the value.

Reply:
I don't understand this feedback,  I'll try to discuss with you via voice

Reply:
Discussed over voice, this PR depends on #8636 .  Now that this PR is rebased on #8636, this is resolved.